File: 2025-9322-team04_finproject_java/src/main/java/Client_Java/player/model/HomepageModel.java
package Client_Java.player.model;

import Client_Java.player.PlayerCorbaManager;

public class HomepageModel {
    private String username;
    private int waitingTime;

    public HomepageModel() {}

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public int getWaitingTime() {
        return waitingTime;
    }

    public void setWaitingTime(int waitingTime) {
        this.waitingTime = waitingTime;
    }

    public boolean isSessionValid() {
        String sessionId = PlayerCorbaManager.getInstance().getSessionId();
        return PlayerCorbaManager.getInstance().getLoginService().isSessionActive(sessionId);
    }
}

File: 2025-9322-team04_finproject_python/Client_Python/player/model/HomepageModel.py
class HomepageModel:
    def __init__(self):
        self.username = ""
        self.waiting_time = 0  # Countdown time in seconds

    def get_username(self):
        return self.username

    def set_username(self, username):
        self.username = username

    def get_waiting_time(self):
        return self.waiting_time

    def set_waiting_time(self, waiting_time):
        self.waiting_time = waiting_time

    def is_session_valid(self, corba_manager):
        session_id = corba_manager.get_session_id()
        return corba_manager.get_login_service().isSessionActive(session_id)